src/main/webapp/user-private-comments.jsp

```diff
@@ -14,27 +13,56 @@
 %>
 
 <%--HTML--%>
-<ul>
-  <%if (currUser) {%>
-    <h1 id="pc-header">Your Private Task Comments:</h1>
-    <%for (TaskData task : tasks) {%>
-      <li class="task-with-pc">
-        <button class="inline deep-button" type="button" onclick="location.href='task?taskID=<%=task.getTaskID()%>'">
-          <%=task.getName()%>
-        </button>
-        <%request.setAttribute("task", task);%>
-        <%request.setAttribute("clickable", false);%>
-        <jsp:include page="task-status-checkmark.jsp"/>
-        <p class="inline"><%=task.getDescription()%></p>
-        <form action="/user-profile" method="POST">
-          <input type="hidden" name="taskID" value="<%=task.getTaskID()%>">
-          <br>
-          <textarea style="width:300px; height:100px;" type="text" name="message-<%=task.getTaskID()%>" rows="6" cols="50"><%=(privateCommentsMap.get(task.getTaskID())).getMessage().trim()%></textarea>
-          <br>
-          <button class="deep-button" style="font-size: 12px;" type="submit" class="deep-button">Update Comment</button>
-        </form>
-        <br><br>
-      </li>
-    <%}%>
-  <%}%>
-</ul>
\ No newline at end of file
+<html>
+  <head>
+    <script src="https://cdn.jsdelivr.net/npm/@editorjs/editorjs@latest"></script>
+    <script src="https://cdn.jsdelivr.net/npm/@editorjs/header@latest"></script>
+  </head>
+  <script>
+    function init() {
+      const editor = new EditorJS({
+        holderId : 'box',
+        autofocus: true,
+        data: jsData,
+        tools: {
+          header: Header,
+        },
+      });
+    }
+    function save() {
+      editor.save().then((outputData) => {
+        console.log('Article data: ', outputData)
+      }).catch((error) => {
+        console.log('Saving failed: ', error)
+      });  
+    }
+  </script> 
+  <body>  
+    <ul>
+      <%if (currUser) {%>
+        <h1 id="pc-header">Your Private Task Comments:</h1>
+        <%for (TaskData task : tasks) {%>
+          <script>var jsData = <%(privateCommentsMap.get(task.getTaskID())).getMessage().trim();%></script>
+          <li class="task-with-pc">
+            <button class="inline deep-button" type="button" onclick="location.href='task?taskID=<%=task.getTaskID()%>'">
+              <%=task.getName()%>
+            </button>
+            <%request.setAttribute("task", task);%>
+            <%request.setAttribute("clickable", false);%>
+            <jsp:include page="task-status-checkmark.jsp"/>
+            <p class="inline"><%=task.getDescription()%></p>
+            <form onload="init()" action="/user-profile" method="POST">
+              <input type="hidden" name="taskID" value="<%=task.getTaskID()%>">
+              <br>
+              <div id="box" type="text"></div>
+              <br>
+              <input type="hidden" name="message-<%=task.getTaskID()%>" value="<%=document.getElementById('box').value(outputData);%>"</input>
```

Comment:
error here: "Invalid character constant". I'm not sure but I think that has to do with "outputData". Perhaps I call this only after submit?

Also I think getting the element ID shouldn't be an issue (because the box id changes for every task in the loop) or should I name the ID box-<%task.getTaskID()%> ?

Reply:
> Also I think getting the element ID shouldn't be an issue (because the box id changes for every task in the loop) or should I name the ID box-<%task.getTaskID()%> ?

the editorjs will need divs with different ids, since "id=box" will match a bunch at once.


> <input type="hidden" name="message-<%=task.getTaskID()%>" value="<%=document.getElementById('box').value(outputData);%>"</input>

I'm not sure about this one.  It's mixing jsp/java (the `<%` tags) with javascript (document.getElementById).  Although you're building them both here at the same time, they are executed separately (jsp/java on the server, javascript on the client)